src/test/java/org/opensearch/knn/index/mapper/KNNVectorFieldMapperTests.java

```diff
@@ -858,20 +861,143 @@ public void testMethodFieldMapperParseCreateField_validInput_thenDifferentFieldT
                 doReturn(Optional.of(TEST_VECTOR)).when(methodFieldMapper).getFloatsFromContext(parseContext, TEST_DIMENSION);
             } else {
                 doReturn(Optional.of(TEST_BYTE_VECTOR)).when(methodFieldMapper)
-                    .getBytesFromContext(parseContext, dataType == VectorDataType.BINARY ? TEST_DIMENSION * 8 : TEST_DIMENSION, dataType);
+                    .getBytesFromContext(
+                        parseContext,
+                        dataType == VectorDataType.BINARY ? TEST_DIMENSION * Byte.SIZE : TEST_DIMENSION,
+                        dataType
+                    );
             }
 
             methodFieldMapper.parseCreateField(parseContext, dimension, dataType);
             fields = document.getFields();
             assertEquals(1, fields.size());
             field1 = fields.get(0);
             assertTrue(field1 instanceof VectorField);
+            assertEquals(Integer.parseInt(field1.fieldType().getAttributes().get(DIMENSION_FIELD_NAME)), dimension);
         }
         // making sure to close the static mock to ensure that for tests running on this thread are not impacted by
         // this mocking
         utilMockedStatic.close();
     }
 
+    @SneakyThrows
+    public void testModelFieldMapperParseCreateField_validInput_thenDifferentFieldTypes() {
+        MockedStatic<KNNVectorFieldMapperUtil> utilMockedStatic = Mockito.mockStatic(KNNVectorFieldMapperUtil.class);
+        ModelDao modelDao = mock(ModelDao.class);
+        ModelMetadata modelMetadata = mock(ModelMetadata.class);
+        MockedStatic<ModelUtil> modelUtilMockedStatic = Mockito.mockStatic(ModelUtil.class);
```

Comment:
I would have done it if it was source. I had a split mind here only because it will structure the code in bad readable shape. Hence I generally avoid try with resources on the tests. But let me change it, as I don't have a strong reason of not using try with resource, as it more of an OCD on the code look and feel.

Reply:
updated to use try with resources.